runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/DoFnOp.java

```diff
@@ -479,15 +481,15 @@ static <T, OutT> CompletionStage<WindowedValue<OutT>> createOutputFuture(
   }
 
   static class FutureCollectorImpl<OutT> implements FutureCollector<OutT> {
-    private final List<CompletionStage<WindowedValue<OutT>>> outputFutures;
     private final AtomicBoolean collectorSealed;
+    private CompletionStage<Collection<WindowedValue<OutT>>> outputFuture;
```

Comment:
output future needs to be thread safe given we have moved away from synchronized list.
Especially across `add(..)` and `addAll(..)`

Reply:
Good catch! I added synchronization there.